### PR TITLE
enable both streams working simultaneously

### PR DESCRIPTION
**Pull request by bonuzzz**
Some changes  to get working high res and low res streams working at same time with settings on web interface.
When both streams are enabled the first one is high res, the second one is low res. Both profiles are enabled and supported in ONVIF
Tested on BFUS camera with BlueIris and Dahua vth panel. I don't have original firmware files, so I edited files directly on camera.

### CODE REVIEW

**Comment by bonuzzz** [comment]
when both streams are enabled, rtsp paths would be:
1080p: rtsp://ip-cam:554/ch0_0.h264
360p: rtsp://ip-cam:8554/ch0_1.h264

**Comment by roleoroleo** [comment]
Hello and thanks for the pr.
I have noticed a potential problem with the watchdog. The check_rtsp function needs to be changed because it cannot detect whether the h264grabber or rRTSPServer processes are blocked. Because there are 2 in memory.
Probably we need to check the pid of the process.

**Comment by bonuzzz** [comment]
agree with you
https://gist.github.com/bonuzzz/08370bf5163a9763b3db681f1a5a5275
this is a draft of variant how to check additional processes.

**Comment by bonuzzz** [comment]
Agree with you.
Tried to modify current watchdog solution, but had only one stream after restart. So I don't have working solution for now.

**Comment by roleoroleo** [comment]
I will try to find a solution...
About the original fw, which suffix?

**Comment by bonuzzz** [comment]
y203c
Experimenting with independent binaries for streams. It works, but I figure out only one advantage: if one stream hangs, script restarts only one without breaking another one.
But what is more interesting, i have the same issue with restarting h264grabber, when only one process after restart killing process. (I kill h264grabber process in ssh to test script).
https://github.com/roleoroleo/yi-hack-MStar/pull/113/files#diff-51edf1daa081dccf4bd69c01ed77baa3R149
it is here. and in the most cases second h264grabber for low stream would be restored.

**Comment by bonuzzz** [comment]
I killed h264grabber process on another untouched by me camera and it didn't restored.
I think it could be a bug in watchdog script.

**Comment by roleoroleo** [comment]
Pull request, thanks.

**Comment by roleoroleo** [comment]
Do you agree if i make some small changes to this pr and then merge?

**Comment by bonuzzz** [comment]
of course! this is just an idea. feel free to rewrite

File: src/static/static/home/yi-hack/script/system.sh [hunk 1]
```diff
@@ -136,27 +136,14 @@ if [[ $HTTPD_PORT != "80" ]] ; then
...
-    if [[ $(get_config RTSP_STREAM) == "high" ]] ; then
-        h264grabber -r high | RRTSP_RES=0 RRTSP_PORT=$RTSP_PORT RRTSP_USER=$USERNAME RRTSP_PWD=$PASSWORD rRTSPServer &
+    if [[ $(get_config RTSP_STREAM) == "high" ]] || [[ $(get_config RTSP_STREAM) == "both" ]]; then
+        h264grabber_h -r high | RRTSP_RES=0 RRTSP_PORT=$RTSP_PORT RRTSP_USER=$USERNAME RRTSP_PWD=$PASSWORD rRTSPServer_h &
         ONVIF_PROFILE_0="--name Profile_0 --width 1920 --height 1080 --url rtsp://%s$D_RTSP_PORT/ch0_0.h264 --snapurl http://%s$D_HTTPD_PORT/cgi-bin/snapshot.sh?res=high --type H264"
     fi
-    if [[ $(get_config RTSP_STREAM) == "low" ]] ; then
-        h264grabber -r low | RRTSP_RES=1 RRTSP_PORT=$RTSP_PORT RRTSP_USER=$USERNAME RRTSP_PWD=$PASSWORD rRTSPServer &
+    if [[ $(get_config RTSP_STREAM) == "low" ]] || [[ $(get_config RTSP_STREAM) == "both" ]]; then
+        h264grabber_l -r low | RRTSP_RES=1 RRTSP_PORT=$RTSP_PORT RRTSP_USER=$USERNAME RRTSP_PWD=$PASSWORD rRTSPServer_l &
```
Comment (bonuzzz) [inline src/static/static/home/yi-hack/script/system.sh]: RTSP1_PORT here. and if you combine stream rules, you have to choose permanent port for high and low streams.
so probably it's better to rewrite and make full independent control for low and high streams.

Comment (roleoroleo) [reply src/static/static/home/yi-hack/script/system.sh]: I think it's better to statically reserve port 554 to the high res and port 8554 for low res.
I will change my commit.

**Comment by roleoroleo** [comment]
Another question.
I don't understand the onvif commit:
https://github.com/roleoroleo/yi-hack-MStar/pull/113/commits/35ac4093e0ea30db77e6e83f2b984fa79917df12

When both streams are enabled the current script passes both profiles to onvif.
Why do you need to select which stream ot use on onvif?

**Comment by bonuzzz** [comment]
because in some situations end device chooses wrong profile and give error. in my case if i try to stream high res stream to dahua nvr, i have error "unsupported video format". but low res stream works perfectly there. if onvif exports both profiles, this is a lottery which profile takes dahua. so i force needed profile on onvif

File: src/static/static/home/yi-hack/script/system.sh (hunk 1 above)
Comment (bonuzzz) [reply src/static/static/home/yi-hack/script/system.sh]: agree

**Comment by bonuzzz** [comment]
probably it's better to write everything from beginning than have a lot of commits. feel free to do it. i don't care about any authorships and mentions.

**Comment by roleoroleo** [comment]
I think you're right but probably this is the last one...

**Comment by roleoroleo** [comment]
> 8554 port is needed here

You need to add the port explicitly only when is different from 554.
If you don't add the port when the value is 8554, onvif_srvd passes the wrong url to the client.

**Comment by bonuzzz** [comment]
ok. probably i don't understand your logic.

**Pull request merged**